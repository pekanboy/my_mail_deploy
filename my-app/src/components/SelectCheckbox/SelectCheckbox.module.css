.container {
    background: #FFFFFF;
    border: 2px solid rgba(223, 226, 231, 0.89);
    box-sizing: border-box;
    border-radius: 4px 0 0 4px;

    width: 124px;
    height: 40px;

    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;

    cursor: pointer;
}

.container:hover {
    background: #E5E5E5;
}

.img {
    width: 25px;
    height: 25px;

    margin-left: 10px;
}

.checkbox {
    width: 20px;
    height: 20px;

    background: #FFFFFF;
    border: 1px solid #000000;
    box-sizing: border-box;
    border-radius: 4px;
}

.text {
    width: 80px;
    height: 20px;

    font-family: "Roboto Light", sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 12px;
    line-height: 14px;
    display: flex;
    align-items: center;
    justify-content: center;

    color: #000000;
}