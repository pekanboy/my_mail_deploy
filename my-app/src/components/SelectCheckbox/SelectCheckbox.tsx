import React from 'react';
import styles from './SelectCheckbox.module.css';

export interface SelectCheckboxProps {
    checked: boolean
    checkHandler: () => void
}

export const SelectCheckbox: React.FC<SelectCheckboxProps> =
    ({checked, checkHandler}) => {
        return (
            <div className={styles.container} onClick={checkHandler}>
                <input
                    className={styles.checkbox}
                    type="checkbox"
                    checked={checked}/>
                <span className={styles.text}>Выбрать все</span>
            </div>
        );
    };
