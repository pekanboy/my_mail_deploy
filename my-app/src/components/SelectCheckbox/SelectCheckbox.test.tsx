import React from 'react';
import {configure, shallow, ShallowWrapper} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {SelectCheckbox, SelectCheckboxProps} from './SelectCheckbox';

configure({adapter: new Adapter()});

const testData = {
    checked: false,
    checkHandler: () => undefined,
} as SelectCheckboxProps;

describe('Test select checkbox component', () => {
    it('to match snapshot ', function() {
        let wrapper: ShallowWrapper = shallow(<SelectCheckbox {...testData} />);
        expect(wrapper).toMatchSnapshot();

        const testDataChecked: SelectCheckboxProps = {
            ...testData,
            checked: true,
        };
        wrapper = shallow(<SelectCheckbox {...testDataChecked} />);
        expect(wrapper).toMatchSnapshot();
    });


    it('test callback called', () => {
        const mockClick = jest.fn();
        const data: SelectCheckboxProps = {
            ...testData,
            checkHandler: mockClick,
        };

        const wrapper: ShallowWrapper = shallow(<SelectCheckbox {...data} />);

        wrapper.find('div').at(0).simulate('click', {});

        expect(mockClick.mock.calls.length).toBe(1);
    });
});
