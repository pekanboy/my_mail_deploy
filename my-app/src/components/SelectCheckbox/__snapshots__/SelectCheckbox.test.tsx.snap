// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test select checkbox component to match snapshot  1`] = `
<div
  className="container"
  onClick={[Function]}
>
  <input
    checked={false}
    className="checkbox"
    type="checkbox"
  />
  <span
    className="text"
  >
    Выбрать все
  </span>
</div>
`;

exports[`Test select checkbox component to match snapshot  2`] = `
<div
  className="container"
  onClick={[Function]}
>
  <input
    checked={true}
    className="checkbox"
    type="checkbox"
  />
  <span
    className="text"
  >
    Выбрать все
  </span>
</div>
`;
